### PR TITLE
4 get api articles article

### PR DESCRIPTION
**Pull request by WolfieKnee**
PR for completed ticket 4 and addressing feedback.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -38,26 +27,72 @@ describe("/api", () => {
...
+				votes: 100,
+				article_img_url:
+					"https://images.pexels.com/photos/158651/news-newsletter-newspaper-information-158651.jpeg?w=700&h=700",
+			};
+			return request(app)
+				.get("/api/articles/1")
+				.expect(200)
+				.then((response) => {
+					const { article } = response.body;
+					console.log(article, "<< article");
```
Comment (nc-poonam-rajput) [inline __tests__/app.test.js:73]: remove logs before making commits!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -38,26 +27,72 @@ describe("/api", () => {
...
+					for (const key in expected) {
+						expect(article.key).toEqual(expected.key);
+					}
```
Comment (nc-poonam-rajput) [inline __tests__/app.test.js:76]: fair enough 🎉

File: __tests__/app.test.js [hunk 3]
```diff
@@ -38,26 +27,72 @@ describe("/api", () => {
...
+					expect(msg).toBe("Bad Request");
+				});
+		});
+		test("GET: 404 /20 should respond with Not Found", () => {
+			return request(app)
+				.get("/api/articles/20")
+				.expect(404)
+				.then((response) => {
+					const { msg } = response.body;
+					expect(msg).toBe("Not Found");
```
Comment (nc-poonam-rajput) [inline __tests__/app.test.js:94]: lovelyyyy

File: app.js [hunk 4]
```diff
@@ -1,34 +1,45 @@
...
+		res.status(404).send({ msg: "Not Found" });
+	} else {
+		next(err);
+	}
+});
+
+app.use((err, req, res, next) => {
 	res.status(500).send({
 		message: "500 error! sorry, Dr Dev made a boo-boo",
 	});
```
Comment (nc-poonam-rajput) [inline app.js:45]: Now your err handling is more complex I'd consider extracting the middleware into named functions to make debugging/navigating way nicer for the err handling

**Review by nc-poonam-rajput — COMMENTED** [review]
Wonderful work David!

File: __tests__/app.test.js (hunk 1 above)
Comment (WolfieKnee) [reply __tests__/app.test.js:73]: added spyOn test

File: app.js (hunk 4 above)
Comment (WolfieKnee) [reply app.js:45]: refactored in next PR

**Pull request merged**